api/STYLE.md

```diff
@@ -169,3 +169,58 @@ metadata. We describe these annotations below by category.
   the given package. This is consumed by `protoxform`.
 * `option (udpa.annotations.file_status).work_in_progress = true;` to denote a
   file that is still work-in-progress and subject to breaking changes.
+
+## Principles
+
+The following principles should be adhered to when extending or modifying the
+xDS APIs:
+
+* The xDS APIs have a logical distinction between transport and data model:
+  - The xDS transport protocol describes the network transport on which xDS
+    configuration resources are delivered to clients. A versioned gRPC streaming
+    protocol with support for ACK/NACK is provided by xDS; this is known as the
+    xDS transport protocol (xDS-TP). xDS configuration resources can also be
+    delivered on other transports, e.g. HTTP or filesystem, with some
+    limitations (e.g. no version feedback).
+  - The xDS data model describes the xDS configuration resources themselves,
+    e.g. listeners, route configurations, clusters, endpoints, secrets.
+
+* The xDS APIs are directionally client and server neutral. While many aspects
+  of the APIs reflect the history of their origin as Envoy's control plane APIs,
+  API decisions going forward should reflect the principle of client neutrality.
+
+* The xDS APIs are expressed canonically as [Proto3](https://developers.google.com/protocol-buffers/docs/proto3).
+  Both JSON and YAML are also supported formats, with the standard JSON-proto3
+  conversion used during client configuration ingestion.
+
+* xDS APIs are eventual consistency first. For example, if RDS references a
+  cluster that has not yet been supplied by CDS, it should be silently ignored
+  and traffic not forwarded until the CDS update occurs. Stronger consistency
+  guarantees are possible if the management server is able to sequence the xDS
+  APIs carefully (for example by using the ADS API below). By following the
+  `[CDS, EDS, LDS, RDS]` sequence for all pertinent resources, it will be
+  possible to avoid traffic outages during configuration update.
+
+* The API is primarily intended for machine generation and consumption. It is
+  expected that the management server is responsible for mapping higher level
```

Comment:
Do you want to add some color here around our conversation the other day? Basically that although the API is primarily intended for machine generation (verbosity is not a primary concern), we do value readability of configuration. IMO this is an important point. WDYT?